packages/create-next-app/templates/index.ts

```diff
@@ -179,69 +177,70 @@ export const installTemplate = async ({
       start: 'next start',
       lint: 'next lint',
     },
+    /**
+     * Default dependencies.
+     */
+    dependencies: {
+      react: 'latest',
+      'react-dom': 'latest',
+      next: process.env.NEXT_PRIVATE_TEST_VERSION ?? 'latest',
+    },
+    devDependencies: {},
   }
-  await fs.promises.writeFile(
-    path.join(root, 'package.json'),
-    JSON.stringify(packageJson, null, 2) + os.EOL
-  )
-
-  /**
-   * These flags will be passed to `install()`, which calls the package manager
-   * install process.
-   */
-  const installFlags = { packageManager, isOnline }
-
-  /**
-   * Default dependencies.
-   */
-  const dependencies = [
-    'react',
-    'react-dom',
-    `next${
-      process.env.NEXT_PRIVATE_TEST_VERSION
-        ? `@${process.env.NEXT_PRIVATE_TEST_VERSION}`
-        : ''
-    }`,
-  ]
 
   /**
    * TypeScript projects will have type definitions and other devDependencies.
    */
   if (mode === 'ts') {
-    dependencies.push(
-      'typescript',
-      '@types/react',
-      '@types/node',
-      '@types/react-dom'
-    )
+    packageJson.devDependencies = {
+      ...packageJson.devDependencies,
+      typescript: 'latest',
+      '@types/react': 'latest',
+      '@types/node': 'latest',
+      '@types/react-dom': 'latest',
+    }
   }
 
-  /**
-   * Add Tailwind CSS dependencies.
-   */
+  /* Add Tailwind CSS dependencies. */
   if (tailwind) {
-    dependencies.push('tailwindcss', 'postcss', 'autoprefixer')
+    packageJson.devDependencies = {
+      ...packageJson.devDependencies,
+      postcss: 'latest',
+      tailwindcss: 'latest',
+    }
   }
 
-  /**
-   * Default eslint dependencies.
-   */
+  /* Default ESLint dependencies. */
   if (eslint) {
-    dependencies.push('eslint', 'eslint-config-next')
-  }
-  /**
-   * Install package.json dependencies if they exist.
-   */
-  if (dependencies.length) {
-    console.log()
-    console.log('Installing dependencies:')
-    for (const dependency of dependencies) {
-      console.log(`- ${cyan(dependency)}`)
+    packageJson.devDependencies = {
+      ...packageJson.devDependencies,
+      eslint: 'latest',
+      'eslint-config-next': 'latest',
     }
-    console.log()
+  }
 
-    await install(root, dependencies, installFlags)
+  await fs.writeFile(
+    path.join(root, 'package.json'),
+    JSON.stringify(packageJson, null, 2) + os.EOL
+  )
+
+  /* Install package.json dependencies */
+  console.log('\nInstalling dependencies:')
+  for (const dependency in packageJson.dependencies)
+    console.log(`- ${cyan(dependency)}`)
+
+  const devDeps = Object.keys(packageJson.devDependencies).length
+  if (devDeps) {
+    console.log('\nInstalling devDependencies:')
+    for (const dependency in packageJson.devDependencies)
+      console.log(`- ${cyan(dependency)}`)
+  } else {
+    delete packageJson.devDependencies
```

Comment:
Whats the purpose of `delete` here?

Reply:
See https://github.com/vercel/next.js/pull/55725

Reply:
just to get rid of an empty object `{}`

Reply:
But it was already stringified on line 224 above so I dont think line 238 is going to do anything